File: tests/rosetta/transpiler/php/arbitrary-precision-integers-included-.php
<?php
function pow_int($base, $exp) {
  global $pow_big, $e1, $e2, $x, $s;
  $result = 1;
  $b = $base;
  $e = $exp;
  while ($e > 0) {
  if ($e % 2 == 1) {
  $result = $result * $b;
}
  $b = $b * $b;
  $e = intval((intdiv($e, 2)));
};
  return $result;
}
function pow_big($base, $exp) {
  global $pow_int, $e1, $e2, $x, $s;
  $result = 1;
  $b = $base;
  $e = $exp;
  while ($e > 0) {
  if ($e % 2 == 1) {
  $result = $result * $b;
}
  $b = $b * $b;
  $e = intval((intdiv($e, 2)));
};
  return $result;
}
$e1 = pow_int(3, 2);
$e2 = pow_int(4, $e1);
$base = 5;
$x = pow_big($base, $e2);
$s = json_encode($x, 1344);
echo "5^(4^(3^2)) has" . " " . json_encode(strlen($s), 1344) . " " . "digits:" . " " . json_encode(substr($s, 0, 20 - 0), 1344) . " " . "..." . " " . json_encode(substr($s, strlen($s) - 20, strlen($s) - strlen($s) - 20), 1344), PHP_EOL;

File: tests/rosetta/transpiler/Kotlin/arbitrary-precision-integers-included-.kt
import java.math.BigInteger

var e1: Int = pow_int(3, 2)
var e2: Int = pow_int(4, e1)
var base: BigInteger = java.math.BigInteger.valueOf(5)
var x: BigInteger = pow_big(base, e2)
var s: String = x.toString()
fun pow_int(base: Int, exp: Int): Int {
    var result: Int = 1
    var b: Int = base
    var e: Int = exp
    while (e > 0) {
        if ((e % 2) == 1) {
            result = result * b
        }
        b = b * b
        e = e / 2.toInt()
    }
    return result
}

fun pow_big(base: BigInteger, exp: Int): BigInteger {
    var result: BigInteger = java.math.BigInteger.valueOf(1)
    var b: BigInteger = base
    var e: Int = exp
    while (e > 0) {
        if ((e % 2) == 1) {
            result = result.multiply(b)
        }
        b = b.multiply(b)
        e = e / 2.toInt()
    }
    return result
}

fun main() {
    println(listOf("5^(4^(3^2)) has", s.length, "digits:", s.substring(0, 20), "...", s.substring(s.length - 20, s.length)).joinToString(" "))
}
